### PR TITLE
Run tests in parallel

### PR DESCRIPTION
**Pull request by mstajdohar**

### CODE REVIEW

**Comment by mstajdohar** [comment]
The coverage appears decreased from 88% to 85%. This is because I have enabled stricter branch checking in the coverage configuration.

**Comment by mstajdohar** [comment]
@dblenkus, this PR is ready for review. Do not merge before I update `resolwe-bio` also.

File: docs/CHANGELOG.rst [hunk 1]
```diff
@@ -36,6 +53,7 @@ Added
   high priority.
 - Add ``resolwe.elastic`` application, a framework for advanced indexing of
   Django models with ElasticSearch
+- Support version in descriptor schema yaml files
```
Comment (dblenkus) [inline docs/CHANGELOG.rst]: Remove this one

File: resolwe/flow/utils/test.py [hunk 2]
```diff
@@ -198,8 +198,42 @@ def _register_schemas(self, **kwargs):
...
+        while True:
+            try:
+                counter = 1
+                for name in os.listdir(path):
+                    if os.path.isdir(os.path.join(path, name)) and name.startswith('test'):
+                        try:
+                            current = int(name.split('_')[-1])
+                            if current >= counter:
+                                counter = current + 1
+                        except:  # pylint: disable=bare-except
```
Comment (dblenkus) [inline resolwe/flow/utils/test.py]: Catch only `ValueError`

File: resolwe/flow/utils/test.py [hunk 3]
```diff
@@ -198,8 +198,42 @@ def _register_schemas(self, **kwargs):
...
+                continue
+
+        return test_data_dir
+
     def setUp(self):
         """Initialize test data."""
+        flow_executor_settings = copy.copy(getattr(settings, 'FLOW_EXECUTOR', {}))
+        flow_executor_settings['DATA_DIR'] = self._test_data_dir(FLOW_EXECUTOR_SETTINGS['DATA_DIR'])
+
+        self.settings = override_settings(FLOW_EXECUTOR=flow_executor_settings)
```
Comment (dblenkus) [inline resolwe/flow/utils/test.py]: Use more descriptive name then `settings`.

File: MANIFEST.in [hunk 4]
```diff
@@ -8,6 +8,7 @@ recursive-include docs/images *.png
 # include tests and files needed by tests
 include tox.ini
 include .pylintrc
+include .coveragerc
```
Comment (dblenkus) [inline MANIFEST.in]: Add this to previous commit.

File: docs/contributing.rst [hunk 5]
```diff
@@ -93,7 +93,7 @@ Running tests
 To run the tests, use::
 
     cd tests
-    ./manage.py test resolwe
+    ./manage.py test resolwe --parallel
```
Comment (dblenkus) [inline docs/contributing.rst]: This also belongs to the previous commit

File: resolwe/elastic/utils/tests.py [hunk 6]
```diff
@@ -1,35 +0,0 @@
...
-from django.test import TestCase, override_settings
-
-from resolwe.elastic.builder import index_builder
-
-
-__all__ = ('ElasticSearchTestCase',)
-
-
-@override_settings(ELASTICSEARCH_INDEX_PREFIX='test')
-class ElasticSearchTestCase(TestCase):
```
Comment (dblenkus) [inline resolwe/elastic/utils/tests.py]: Leave this one in elastic app, to have all ES stuff in one place

**Comment by mstajdohar** [comment]
> Would resolwe.flow.test be better place for this module? Because resolwe.flow is our main app and test utils should be probably included.

I would prefer to keep the test framework in one place.

**Comment by kostko** [comment]
> I would prefer to keep the test framework in one place.

I agree with @mstajdohar on this one, `resolwe.test` is a better choice here.

**Comment by mstajdohar** [comment]
@dblenkus, I have addressed you comments.

**Pull request merged**